llvm/lib/Target/AArch64/AArch64LoadStoreOptimizer.cpp

```diff
@@ -2159,6 +2171,24 @@ bool AArch64LoadStoreOpt::tryToPairLdStInst(MachineBasicBlock::iterator &MBBI) {
     // Keeping the iterator straight is a pain, so we let the merge routine tell
     // us what the next instruction is after it's done mucking about.
     auto Prev = std::prev(MBBI);
+
+    // Get the needed alignments to check them if
+    // ldp-aligned-only/stp-aligned-only features are opted.
+    uint64_t MemAlignment = MemOp->getAlign().value();
```

Comment:
This looks like it could be performed higher, up above the call to findMatchingInsn? So long as it doesn't need to check the second instruction in the pair too.

If so there are some other conditions checked in isCandidateToMergeOrPair. I'm not sure why they are separated out, but it might be good to keep those kinds of checks together in the same function.

Reply:
I think I will move it right under the call to isCandidateToMergeOrPair and not inside it, because I do not want to alter the outcome of the function for merges (merge != pair).

Reply:
This should remain where it is, because it breaks in case "if (Paired != E)" is not entered and the function is going to "return false".